### PR TITLE
V5.1.0

### PR DESCRIPTION
**Pull request by srudaa**
### Changed

- Custom TCP connector is switched out to use the Sockets module in the framework. 
- Alpine docker image is building binary instead. This reduces the image size. 

### Fixed

- Able to deal with asterisk chars in the payload.

### CODE REVIEW

File: Source/SentenceParser.cs [hunk 1]
```diff
@@ -12,20 +12,21 @@ namespace RaaLabs.Edge.Connectors.Terasaki
...
             // Calculate the checksum by folding over the sentence up to the asterisk (*) character, using XOR
-            byte calculatedChecksum = sentence.TakeWhile(c => c != '*').Aggregate((byte)0x00, (sum, next) => (byte)(sum ^ next));
-
             // Extract the different parts of the sentence using regex
             var match = _sentencePattern.Match(sentence);
             ThrowIfUnmatchedPattern(match, sentence);
 
             var (formatIdentifier, payload, checksumStr, _) = match.Groups.Values.Skip(1).Select(g => g.Value).ToList();
+
+            var calculatedChecksum = sentence.Substring(0, sentence.LastIndexOf("*", StringComparison.Ordinal)).Aggregate((byte)0x00, (sum, next) => (byte)(sum ^ next));
```
Comment (rafaelschlatter) [inline Source/SentenceParser.cs:29]: `state` is not used and can be named `_` (line 47, but cant add lines to code that is not in diff -.-)

**Review by rafaelschlatter — CHANGES_REQUESTED** [review]
Nice, code is simpler with the Sockets module. One inline comment. PR needs a label to create a new version (will be 4.3.0 if minor, 4.2.0 already exists).

**Comment by rafaelschlatter** [comment]
@srudaa should we include batching in this PR?

**Comment by srudaa** [comment]
> @srudaa should we include batching in this PR?

No, let's create a separate one for that.

**Review by rafaelschlatter — APPROVED** [review]

**Comment by rafaelschlatter** [comment]
@thotokraa could you take a look at this PR?

**Pull request merged**